### ISSUE TITLE
Recurrent Batch Normalization

### ISSUE THREAD

**Issue by David-Taub** [keras-team/keras#10193]
[This paper](https://arxiv.org/pdf/1603.09025.pdf) (2017, Cooijmans et al.) suggests a batch normalization on 2 of the transformation of the LSTM, which are internal to the LSTM cell. This sometimes makes models train faster and generalize better.

**Can someone think of an easy implemention for such a normalization?**

As far as I can tell, a user implemention for this would include writing a custom LSTM layer. I don't know what's the standard for inserting such methods into the Keras codebase, but IMO this could be useful for many people. If there is no easy implementation and it is decided to be in Keras, a possible fix is to add this in the LSTM cell and enable this normalization by a batch_normalization boolean argument to the LSTM layer & cell.

@fchollet do you think an implementation would be inserted into Keras? If so I can try make a PR.

**Comment by elisim** [comment]
@fchollet any progress?

**Comment by MLightM** [comment]
@fchollet any progress?
